decider-sequenza/src/main/java/ca/on/oicr/pde/deciders/SequenzaDecider.java

```diff
@@ -102,19 +134,78 @@ protected ReturnValue doFinalCheck(String commaSeparatedFilePaths, String commaS
                 if (!bs.getPath().equals(p)) {
                     continue;
                 }
-                String tt = bs.getTissueType();
+                String tt = bs.tissueType;
+                if (!tt.isEmpty() && !tt.equals("R")){
+                    extTum = bs.getRootSampleName();
+                    this.tumourFilePath = p;
+                    match.add(p); // GP-1627:first element of match array list contains tumour bam file path
+                    haveTumr = true; 
+                    break;
+                }
+            }
+        }
 
-                if (!tt.isEmpty() && tt.equals("R")) {
-                    haveNorm = true;
-                } else if (!tt.isEmpty()) {
-                    haveTumr = true;
+        // Check for duplicate file names and exclude them from analysis
+        this.duplicates = detectDuplicates(commaSeparatedFilePaths);
+
+        for (String p : filePaths) {
+            if (null != this.duplicates && this.duplicates.contains(p)) {
+                Log.stderr("File [" + p + "] has a name that cannot be disambiguated in current set, will skip it");
+                continue;
+            }
+            if (p != this.tumourFilePath) { // check if this path is not tumour bam
+
+                for (BeSmall bs : fileSwaToSmall.values()) {
+                    if (!bs.getPath().equals(p)) {
+                        continue;
+                    }
+                    if (bs.getPath().equals(this.tumourFilePath)){ // check if this path is not tumour bam
+                        continue;
+                    }
+                    
+                    String rootSampleName = bs.getRootSampleName();
+                    String tt = bs.getTissueType();
+                    // GP-1627: update match list with normal bam if, root-sample-name for normal == tumour 
+                    if (!tt.isEmpty() && tt.equals("R") && rootSampleName.equals(extTum)) { 
```

Comment:
Instead of hard-coding normal 'R' I would have moved it into a static variable so it is easier to maintain

Reply:
I won't comment on other occurrences of 'R'